<?xml version="1.0" ?>
<launch>
  <!--  -->
  <include file="$(find mrn_kinect_bin)/launch/kinect_demo.launch" ></include>

  <arg name="robot_ips" default="172.16.0.104 172.16.0.103"/>
  <arg name="tracking_robot_ip" default="172.16.0.104"/>
  <arg name="impedance_robot_ip" default="172.16.0.103"/>

  <arg name="load_gripper" default="true" />
  <include file="$(find franka_control)/launch/franka_control.launch" >
    <arg name="robot_ip" value="$(arg robot_ips)" />
    <arg name="load_gripper" value="$(arg load_gripper)" />
  </include>

  <rosparam command="load" file="$(find franka_panda_controller_swc)/config/franka_panda_controller_swc.yaml" />
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="cartesian_impedance_controller"/>
  <!-- node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find franka_panda_controller_swc)/launch/rviz/franka_description_with_marker.rviz"/ -->
  <node name="interactive_marker" pkg="mrn_panda" type="hand_tracking.py" required="true" output="screen">
    <param name="link_name" value="panda_link0" />
  </node>
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" required="false" />


  <include file="$(find mrn_aruco)/launch/aruco_detect_kinect.launch" ></include>
  <!-- <include file="$(find mrn_kinect_bin)/launch/kinect_demo.launch" ></include> -->

  <node name="mrn_openpose_demo" pkg="mrn_openpose" type="mrn_openpose_demo" args="/registered_097377233947/image_raw /undistorted_097377233947/image_raw 097377233947" />
  <!-- <node name="mrn_openpose_demo" pkg="mrn_openpose" type="mrn_openpose_demo" args="/registered_226287140347/image_raw /undistorted_226287140347/image_raw 226287140347" /> -->
  <!-- <include file="$(find mrn_kinect_bin)/launch/kinect_demo.launch" ></include> -->

  <rosparam command="load" file="$(find panda_ros)/config/franka_panda_controller_swc.yaml" />
  <node name="controller_spawner2" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="cartesian_impedance_equilibrium_controller"/>
  <node name="equilibrium_publisher" pkg="panda_ros" type="equilibrium_publisher.py" required="true" >
    <param name="link_name" value="panda_link0" />
  </node>
</launch>
